dstep/translator/CodeBlock.d

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright: Copyright (c) 2016 Wojciech Szęszoł. All rights reserved.
+ * Authors: Wojciech Szęszoł
+ * Version: Initial created: Mar 11, 2016
+ * License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0)
+ */
+
+module dstep.translator.CodeBlock;
+
+import std.array;
+import std.string;
+
+enum EndlHint
+{
+    Empty, // does nothing
+    Group,
+    SingleLine,
+    MultiLine, // indent children
+    SubscopeWeak, // put braces if children takes more than single line
+    SubscopeStrong, // always put braces
+}
+
+struct CodeBlock
+{
+    string spelling;
+    EndlHint endlHint;
+    CodeBlock[] children;
+
+    this(string spelling, EndlHint endlHint, CodeBlock[] children = [])
+    {
+        this.spelling = spelling;
+        this.endlHint = endlHint;
+        this.children = children;
+    }
+
+    this(EndlHint endlHint, CodeBlock[] children = [])
+    {
+        this.spelling = "";
+        this.endlHint = endlHint;
+        this.children = children;
+    }
+
+    this(string spelling)
+    {
+        this.spelling = spelling;
+        this.endlHint = EndlHint.SingleLine;
+        this.children = [];   
+    }
+
+    this(CodeBlock[] children)
+    {
+        this.spelling = null;
+        this.endlHint = EndlHint.Group;
+        this.children = children;
+    }
+}
```

Comment:
Doesn't matter that much. I was just thinking since D has so flexible struct initialization syntax.